### PR TITLE
Issue #10719: Intent targeting the current app is not an app link

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt [hunk 1]
```diff
@@ -100,6 +101,12 @@ class AppLinksInterceptor(
         val result = handleRedirect(redirect, uri)
 
         if (redirect.isRedirect()) {
+            if (redirect.appIntent?.`package` != null &&
```
Comment (csadilek) [inline components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:104]: Could we handle this in `AppLinksUseCases.safeParseUri` instead and return null there. Then we basically have the same fix as in Fennec, plus we don't add more conditions here. This is already quite hard to follow :).

File: components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt [hunk 2]
```diff
@@ -265,7 +265,7 @@ class AppLinksInterceptorTest {
     }
 
     @Test
-    fun `blocklisted schemes request always ignored even if the engine does not support it`() {
+    fun `blacklisted schemes request always ignored even if the engine does not support it`() {
```
Comment (csadilek) [inline components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt:268]: Let's either use `Denied schemes` or `Blocked schemes`.

File: components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt [hunk 3]
```diff
@@ -285,6 +285,29 @@ class AppLinksInterceptorTest {
         assertEquals(null, response)
     }
 
+    @Test
+    fun `request target blacklisted package is always ignored even if the engine does not support it`() {
```
Comment (csadilek) [inline components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt:289]: Same here: Block list or denied package.

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt (hunk 1 above)
Comment (rocketsroger) [reply components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:104]: Understood.  I'll move the implementation into `safeParseUri`.  Thanks

**Review by csadilek — APPROVED** [review]
🚢
